src/predict.jl

```diff
@@ -0,0 +1,495 @@
+
+    if use_re && newX === m.X
+        # should we add a kwarg to fitted to allow returning eta?
+        type == :response && return fitted(m)
+        return m.resp.eta
+    end
+    y = zeros(T, nobs(m))
+    predict!(y, m, newX; use_re=use_re, type=type)
+end
+
+"""
+
+StatsBase.predict(m::LinearMixedModel{T}, newdata;
+                  new_re_levels=:missing)
+StatsBase.predict(m::GeneralizedLinearMixedModel{T}, newdata;
+                  new_re_levels=:missing, type=:response)
+
+Predict response for new values of the fixed-effects matrix X.
+
+!!! note
+    Currently, no in-place methods are provided because these methods
+    internally construct a new model and therefore allocate not just a
+    response vector but also many other matrices.
+
+!!! warning
+    Models are assumed to be full rank.
+
+!!! warning
+    These methods construct an entire MixedModel behind the scenes and
+    as such may use a large amount of memory when `newdata` is large.
+
+The keyword argument `new_re_levels` specifies how previously unobserved
+values of the grouping variable are handled. Possible values are
+`:population` (return population values, i.e. fixed-effects only),
+`:missing` (return `missing`), `:error` (error on this condition;
+the error type is an implementation detail), `:simulate` (simulate new values).
+For `:simulate`, the values are determined by solving for their values by
+using the existing model's estimates for the new data. (These are in general
+*not* the same values as the estimates computed on the new data.)
+
+Predictions based purely on the fixed effects can be obtained with by
+specifying previously unobserved levels of the random effects and setting
+`new_re_levels=:population`. In the future, it may be possible to specify
+a subset of the grouping variables or overall random-effects structure to use,
+but not at this time.
+
+For [`GeneralizedLinearMixedModel`](@ref), the `type` parameter specifies
+whether the predictions should be returned on the scale of linear predictor
+(`:linpred`) or on the response scale (`:response`). If you don't know the
+difference between these terms, then you probably want `type=:response`.
+
+Regression weights are not yet supported in prediction. As a consequence of this,
+`new_re_levels=:simulate` is also not yet available for `GeneralizedLinearMixedModel`.
+Similarly, offset are also not support for `GeneralizedLinearMixedModel`.
+
+!!! note
+    The `predict` and `predict!` methods with `newX` as a fixed effects matrix
+    differ from the methods with `newdata` in tabular format. The matrix
+    methods are more efficient than the tabular methods. The tabular methods
+    can accomodate new values of the grouping variable(s), but the matrix methods
+    cannot.
+"""
+function StatsBase.predict(m::LinearMixedModel{T}, newdata::Tables.ColumnTable;
+                           new_re_levels=:population) where T
+
+    new_re_levels in (:population, :missing, :)
```

Comment:
Is there a missing `|| throw(...` here?

Reply:
yes -- looks like a chunk got chopped off at some point.